components/datadog/agent/params.go

```diff
@@ -70,18 +83,33 @@ func WithVersion(version string) func(*Params) error {
 	}
 }
 
-// WithPipelineID use a specific version of the Agent by pipeline id. For example: `16497585` uses the version `pipeline-16497585`
-func WithPipelineID(version string) func(*Params) error {
+// WithPipelineID uses a specific testing pipeline ID of the datadog-agent CI. For example: `16497585`
+func WithPipelineID(pipelineID string) func(*Params) error {
 	return func(p *Params) error {
-		p.version = parsePipelineVersion(version)
+		p.version.Repository = os.TestingRepository
+		p.version.PipelineID = pipelineID
+		return nil
+	}
+}
 
+// WithRepository uses a specific repository of the Agent. For example: `staging` or `trial`
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Not a blocker: can you add the default value ?